tiledb/sm/query/readers/aggregators/min_max_aggregator.cc

```diff
@@ -0,0 +1,413 @@
+/**
+ * @file min_max_aggregator.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2023 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file implements class MinMaxAggregator.
+ */
+
+#include "tiledb/sm/query/readers/aggregators/min_max_aggregator.h"
+
+#include "tiledb/sm/array_schema/array_schema.h"
+#include "tiledb/sm/query/query_buffer.h"
+#include "tiledb/sm/query/readers/aggregators/aggregate_buffer.h"
+
+using namespace tiledb::common;
+
+namespace tiledb {
+namespace sm {
+
+class MinMaxAggregatorStatusException : public StatusException {
+ public:
+  explicit MinMaxAggregatorStatusException(const std::string& message)
+      : StatusException("MinMaxAggregator", message) {
+  }
+};
+
+template <typename T>
+MinMaxAggregator<T>::MinMaxAggregator(
+    const bool min,
+    const std::string field_name,
+    const bool var_sized,
```

Comment:
These parameter lists everywhere are really hard to read. Please add a `struct FieldInfo` (rename as you like) to hold this information, and start using it here. There are [[several](https://github.com/single-cell-data/TileDB-SOMA/blob/e464b0f11b420c7a0993d383efa25f8168d2ecda/libtiledbsoma/src/soma/column_buffer.h#L277-L314)] [[examples](https://github.com/TileDB-Inc/TileDB-Py/blob/758ebac83ecae6a26a8aeb8894157a5a0ef65a1c/tiledb/py_arrow_io_impl.h#L117-L136)] in API client code, and we should do the same internally.